--- a/ltconfig	2026-06-10 00:26:16.524448084 -0500
+++ b/ltconfig	2026-06-10 00:26:16.559755722 -0500
@@ -457,6 +457,11 @@
 host_vendor=`echo $host | sed 's/^\([^-]*\)-\([^-]*\)-\(.*\)$/\2/'`
 host_os=`echo $host | sed 's/^\([^-]*\)-\([^-]*\)-\(.*\)$/\3/'`
 
+# Substrate is ELF + GNU ld + Linux-like; reuse the linux-gnu profile.
+case "$host_os" in
+substrate*) host=`echo $host | sed 's/-substrate/-linux-gnu/'`; host_os=linux-gnu ;;
+esac
+
 case "$host_os" in
 aix3*)
   # AIX sometimes has problems with the GCC collect2 program.  For some
