diff --git a/src/player.cpp b/src/player.cpp
index 7f438bec..867fb108 100644
--- a/src/player.cpp
+++ b/src/player.cpp
@@ -666,7 +666,7 @@ void Player::handleDoSetLoopModeEvent(LoopMode mode) {
     m_loop_mode = mode;
     std::string toastMsg = "Loop: ";
     std::string mprisStatus;
-    PsyMP3::MPRIS::LoopStatus mprisEnum = PsyMP3::MPRIS::LoopStatus::None;
+    [[maybe_unused]] PsyMP3::MPRIS::LoopStatus mprisEnum = PsyMP3::MPRIS::LoopStatus::None;
     switch(m_loop_mode) {
         case LoopMode::None:
             toastMsg += "None";
